enhancements/aiops/20230216-enable-gpu-services-training.md

```diff
@@ -0,0 +1,70 @@
+# Title: 
+Enable GPU Services upon submission of initial training job for workload logs.
+
+## Summary: 
+Currently, when a user would like to enable AI services, they will go to the Opni Admin Dashboard. First, they are required to enable Logging and once that is done, they will go to the AIOps panel and check the "Enable GPU Services button". When they hit the "Save" button, the GPU services are installed within the Kubernetes cluster. This includes the workload DRAIN service, the training controller service, the GPU Controller service and CPU Inferencing service. This UX can be avoided by simply detecting the availability of a GPU within the cluster and when the user creates or updates the workload log anomaly watchlist to train a Deep Learning model, that is when these GPU services should be installed, rather than through a checkbox button.
+
+## Use case: 
+This will remove the "Enable GPU Services" check box and now will install the GPU services when the user decides to update the watchlist for the very first time with workloads. Opni GPU services will automatically come up upon the creation or update of a workload log anomaly watchlist.
+
```

Comment:
How would a user enable the GPU services again if they choose to disable them?  They have to update the watchlist again?  Would it instead be feasible to keep the enable button, but still have it automatically done when a user creates/updates the workload watchlist?

We also need to consider the upgrade case i.e a user updates the version of Opni, should we automatically update the version of services running, or should we surface this to the user and let them manually update the services.

Reply:
Yeah, so the plan is if they disable the GPU services, then they'd need to update the watchlist again to have the GPU services enabled. Since it would be necessary to install the GPU services if they only update the watchlist, I think we don't need to keep the enable button but we can discuss that some more.

As for the upgrading of Opni, for now we can surface this to the user.